src/Tribe/Change_Authority/Base.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * Class Tribe__Change_Authority__Base
+ *
+ * The basic general purpose change authority to propagate data from a generic source to a generic destination.
+ */
+abstract class Tribe__Change_Authority__Base implements Tribe__Change_Authority__Interface {
+	/**
+	 * An array defining the name of the fields that should be propagated.
+	 * @var array
+	 */
+	protected $fields = array();
+	/**
+	 * An array of callbacks that will be called when evaluating the propagation to a field.
+	 *
+	 * @var array
+	 */
+	protected $propagation_conditions = array();
+
+	/**
+	 * Sets the fields the change authority should apply a propagation policy to.
+	 *
+	 * @param array $fields
+	 *
+	 * @return mixed
+	 */
+	public function set_fields( array $fields ) {
+		$this->fields = $fields;
+	}
+
+	/**
+	 * Propagates the changes from the source to the destination.
+	 *
+	 * @param mixed $from The source object or data.
+	 * @param mixed $to   The destination object or data.
+	 *
+	 * @return array An associative array in the format [ <field> => <propagated> ]
+	 */
+	public function propagate( $from, $to ) {
+		if ( empty( $this->fields ) ) {
+			return array();
+		}
+
+		if ( empty( $from ) || empty( $to ) ) {
+			return array_combine( $this->fields, array_fill( 0, count( $this->fields ), false ) );
+		}
+
+		$result = array();
+
+		foreach ( $this->fields as $field ) {
+			if ( ! $this->should_propagate( $from, $to, $field ) ) {
+				$result[ $field ] = false;
+				continue;
+			}
+			$result[ $field ] = $this->propagate_field( $from, $to, $field );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Whether a field should be propagated from the source to the destination.
+	 *
+	 * @param mixed  $from The source object or data.
+	 * @param mixed  $to   The destination object or data.
+	 * @param string $type The name of the field that's to be evaluated for propagation.
```

Comment:
Replace `$type` with `$field` to match the method signature? (Ditto in the docblock for the matching interface definition)